src/device_impl.rs

```diff
@@ -72,17 +72,17 @@ where
     /// least 500us after the receiving the first `WouldBlock` error before
     /// calling again to continue.
     pub fn restart_nonblocking(&mut self) -> nb::Result<(), Error<E>> {
-        let mode1 = self.read_register(Register::MODE1)?;
+        let mode1 = self.read_register(Register::MODE1).map_err(nb::Error::Other)?;
```

Comment:
Did you get a compiler error here? I think [this](https://github.com/rust-embedded/nb/blob/075ecd33c53bf8af9d1b584dda7a7669aedf2c50/src/lib.rs#L232) should be used here and wonder why the method would compile at all otherwise.
There are also [tests](https://github.com/eldruin/pwm-pca9685-rs/blob/9db5039015d81fb7565caa680c29d82e2fcc6c11/tests/restart.rs#L106) for this method.

Reply:
this is just an explicit call to construct `nb::Error::Other` rather than implict with `.into()`? does the same thing in a _slightly_ more explicit manner, compiles fine and tests pass.